src/main/java/com/google/devtools/build/lib/analysis/actions/StrippingPathMapper.java

```diff
@@ -0,0 +1,268 @@
+      return Pattern.compile(outputRoot + "/[\\w_-]+/");
+    }
+
+    public String strip(String str) {
+      return pattern.matcher(str).replaceAll(outputRoot + "/");
+    }
+  }
+
+  /**
+   * Is this a strippable path?
+   *
+   * @param artifact   artifact whose path to check
+   * @param outputRoot the output tree's execPath-relative root (e.g. "bazel-out")
+   */
+  private static boolean isOutputPath(ActionInput artifact, PathFragment outputRoot) {
+    // We can't simply check for DerivedArtifact. Output paths can also appear, for example, in
+    // ParamFileActionInput and ActionInputHelper.BasicActionInput.
+    return isOutputPath(artifact.getExecPath(), outputRoot);
+  }
+
+  /** Private utility method: Is this a strippable path? */
+  private static boolean isOutputPath(PathFragment pathFragment, PathFragment outputRoot) {
+    return pathFragment.startsWith(outputRoot);
+  }
+
+  /**
+   * Is this action safe to strip?
+   *
+   * <p>This is distinct from whether we <b>should</b> strip it. An action is stripped if a) the
+   * action is explicitly supported (see {@link PathMappers#SUPPORTED_MNEMONICS) and b) it's safe to
+   * do that (for example, the action doesn't have two inputs in different configurations that would
+   * resolve to the same path if prefixes were removed).
+   *
+   * <p>This method checks b).
+   */
+  private static boolean isPathStrippable(
+      NestedSet<? extends ActionInput> actionInputs, PathFragment outputRoot) {
+    // For qualifying action types, check that no inputs or outputs would clash if paths were
+    // removed, e.g. "bazel-out/k8-fastbuild/foo" and "bazel-out/host/foo".
+    //
+    // A more clever algorithm could remap these with custom prefixes - "bazel-out/1/foo" and
+    // "bazel-out/2/foo" - if experience shows that would help.
+    //
+    // Another approach could keep host paths intact (since the "host" path prefix doesn't vary
+    // with configurations). While this would help more action instances qualify, it also blocks
+    // caching the same action in host and target configurations. This could be mitigated by
+    // stripping the host prefix *only* when the entire action is in the host configuration.
+    HashSet<PathFragment> rootRelativePaths = new HashSet<>();
+    for (ActionInput input : actionInputs.toList()) {
+      if (!isOutputPath(input, outputRoot)) {
+        continue;
+      }
+      // For "bazel-out/k8-fastbuild/foo/bar", get "foo/bar".
+      if (!rootRelativePaths.add(input.getExecPath().subFragment(2))) {
+        // TODO(bazel-team): don't fail on duplicate inputs, i.e. when the same exact exec path
+        // (including config prefix) is included twice.
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /*
+   * Strips the configuration prefix from an output artifact's exec path.
+   */
+  private static PathFragment strip(PathFragment execPath) {
+    return execPath.subFragment(0, 1).getRelative(execPath.subFragment(2));
```

Comment:
I modified this to 

```
private static PathFragment strip(PathFragment execPath) {
    String config = "/" + execPath.subFragment(1, 2).getPathString();
    String mappedPath = execPath.getPathString().replace(config, "");
    return PathFragment.create(mappedPath);
 }
```

to take care of paths in the form `blaze-out/k8-fasbuilds/bin/foo/blaze-out/k8-fasbuilds/genfiles/bar`. The existing one will make the path look like `blaze-out/bin/foo/blaze-out/k8-fasbuilds/genfiles/bar` . Let me know if you have any concern

Reply:
No concern, but I would be very interested in learning where paths like this show up.